lib/parachain/types.go

```diff
@@ -0,0 +1,407 @@
+	*va = CoreState(vdt)
+	return nil
+}
+
+// Value returns the value from the underlying VaryingDataType
+func (va *CoreState) Value() (scale.VaryingDataTypeValue, error) {
+	vdt := scale.VaryingDataType(*va)
+	return vdt.Value()
+}
+
+// NewCoreStateVDT returns a new CoreState VaryingDataType
+func NewCoreStateVDT() (scale.VaryingDataType, error) {
+	vdt, err := scale.NewVaryingDataType(OccupiedCore{}, ScheduledCore{}, Free{})
+	if err != nil {
+		return scale.VaryingDataType{}, fmt.Errorf("create varying data type: %w", err)
+	}
+
+	return vdt, nil
+}
+
+// NewAvailabilityCores returns a new AvailabilityCores
+func NewAvailabilityCores() (scale.VaryingDataTypeSlice, error) {
+	vdt, err := NewCoreStateVDT()
+	if err != nil {
+		return scale.VaryingDataTypeSlice{}, fmt.Errorf("create varying data type: %w", err)
+	}
+
+	return scale.NewVaryingDataTypeSlice(vdt), nil
+}
+
+// UpwardMessage A message from a parachain to its Relay Chain.
+type UpwardMessage []byte
+
+// OutboundHrmpMessage is an HRMP message seen from the perspective of a sender.
+type OutboundHrmpMessage struct {
+	Recipient uint32 `scale:"1"`
+	Data      []byte `scale:"2"`
+}
+
+// ValidationCode is Parachain validation code.
+type ValidationCode []byte
+
+// headData is Parachain head data included in the chain.
+type headData []byte
+
+// CandidateCommitments are Commitments made in a `CandidateReceipt`. Many of these are outputs of validation.
+type CandidateCommitments struct {
+	// Messages destined to be interpreted by the Relay chain itself.
+	UpwardMessages []UpwardMessage `scale:"1"`
+	// Horizontal messages sent by the parachain.
+	HorizontalMessages []OutboundHrmpMessage `scale:"2"`
+	// New validation code.
+	NewValidationCode ValidationCode `scale:"3"`
+	// The head-data produced as a result of execution.
+	HeadData headData `scale:"4"`
+	// The number of messages processed from the DMQ.
+	ProcessedDownwardMessages uint32 `scale:"5"`
+	// The mark which specifies the block number up to which all inbound HRMP messages are processed.
+	HrmpWatermark uint32 `scale:"6"`
+}
+
+// SessionIndex is a session index.
+type SessionIndex uint32
+
+// CommittedCandidateReceipt A candidate-receipt with commitments directly included.
+type CommittedCandidateReceipt struct {
+	// The candidate descriptor.
+	Descriptor CandidateDescriptor `scale:"1"`
+	// The commitments made by the parachain.
+	Commitments CandidateCommitments `scale:"2"`
+}
+
+// AssignmentID The public key of a keypair used by a validator for determining assignments
+// to approve included parachain candidates.
+type AssignmentID [sr25519.PublicKeyLength]byte
+
+// IndexedValidator A validator with its index.
+type IndexedValidator struct {
+	Index     []ValidatorIndex `scale:"-"`
+	Validator []ValidatorID    `scale:"2"`
+}
+
+// IndexedValidatorGroup A validator group with its group index.
+type IndexedValidatorGroup struct {
+	GroupIndex []GroupIndex     `scale:"1"`
+	Validators []ValidatorIndex `scale:"2"`
+}
```

Comment:
I can't find corresponding rust structs, can you point me to them?

Reply:
These aren't used. I'm removing them